front/app/modules/commercial/analytics/admin/hooks/useVisitorReferrerTypes/typings.ts

```diff
@@ -14,7 +14,7 @@ export interface Response {
 export interface TrafficSourcesRow {
   count: number;
   'dimension_referrer_type.id': string;
-  first_dimension_referrer_type_name: ReferrerTypeName;
+  first_dimension_referrer_type_name: ReferrerTypeName | string;
```

Comment:
Since we're not 100% sure what we're getting back here from matomo, I added `string` as a possible type just in case we get some string back we weren't expecting

Reply:
Matomo is not super clear about this but it is always treating the referrer type as a fix set of possible values.
In the [Database Schema](https://github.com/matomo-org/matomo/blob/4.x-dev/plugins/Referrers/Referrers.php#L140) `referrer_type` has only four possible values (is not including social networks)
But in the code of the default Referrer plugin [it is including it](https://github.com/matomo-org/matomo/blob/4.x-dev/plugins/Referrers/Referrers.php#L140)
The `keywords` thing that was confusing us is a variable related to search engines type, is the keyword being searched on the search engine that lead the user to our page.
So we can be sure that there would not be any other string that the ones on `ReferrerType`